### PR TITLE
Add back some helpers used by xmtp-rust-swift, expose via ffi

### PR DESCRIPTION
**Pull request by nakajima**
This is the first step in being able to remove https://github.com/xmtp/xmtp-rust-swift from the iOS SDK.

The longer term goal is to remove https://github.com/xmtp/xmtp-rust-swift in favor of https://github.com/xmtp/libxmtp-swift, which will give us a beachhead in incorporating more of libxmtp's features in the iOS SDK (like v3).

### CODE REVIEW

**Review by richardhuaaa — APPROVED** [review]
LGTM, just some small nits! Small disclaimer - I didn't review the `xmtp_v2` crate, as I assume it's identical to what's in the v2 branch.

Do you have any thoughts about pros/cons of placing the v2 stuff in a separate crate/binary from v3 and pulling both into `libxmtp-swift`, vs placing them in the same binary?

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-    "rust-analyzer.linkedProjects": ["bindings_ffi/Cargo.toml", "examples/cli/Cargo.toml"]
-}
\ No newline at end of file
+  "rust-analyzer.linkedProjects": [
+    "bindings_ffi/Cargo.toml",
+    "examples/cli/Cargo.toml",
+    "./xmtp_v2/Cargo.toml"
```
[the remark this answers is not in the document]
Comment (nakajima) [reply .vscode/settings.json:5]: No I don't think so, good catch.

File: bindings_ffi/src/v2.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    async fn test_recover_public_key_keccak256() {
+        // This test was generated using Etherscans Signature tool: https://etherscan.io/verifySig/18959
+        let addr = "0x1B2a516d691aBb8f08a75B2C73c95c62A1632431";
+        let msg = "TestVector1";
+        let sig_hash = "19d6bec562518e365d07ba3cce26d08a5fffa2cbb1e7fe03c1f2d6a722fd3a5e544097b91f8f8cd11d43b032659f30529139ab1a9ecb6c81ed4a762179e87db81c";
+
+        let sig_bytes = ethers_core::utils::hex::decode(sig_hash).unwrap();
+        let sig = xmtp_cryptography::signature::RecoverableSignature::Eip191Signature(sig_bytes);
+
+        let recovered_addr = sig.recover_address(msg).unwrap();
```
[the remark this answers is not in the document]
Comment (nakajima) [reply bindings_ffi/src/v2.rs]: Fixed in https://github.com/xmtp/libxmtp/pull/229/commits/f386d95c4693a22447e5f1f0c9d25dc576a7cff7.

File: bindings_ffi/src/v2.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#[uniffi::export]
+pub fn diffie_hellman_k256(
+    private_key_bytes: Vec<u8>,
+    public_key_bytes: Vec<u8>,
+) -> Result<Vec<u8>, GenericError> {
+    let shared_secret = xmtp_v2::k256_helper::diffie_hellman_byte_params(
+        private_key_bytes.as_slice(),
+        public_key_bytes.as_slice(),
+    )
+    .map_err(|e| e)?;
```
[the remark this answers is not in the document]
Comment (nakajima) [reply bindings_ffi/src/v2.rs:26]: Nope. This is just me not knowing Rust very well.

**Comment by nakajima** [comment]
Hey @richardhuaaa, would you mind giving me another round of review here just to make sure things look alright?

**Review by richardhuaaa — APPROVED** [review]
LGTM! Curious about the answer to my question from before, but not strongly opinionated.

**Pull request merged**